zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,904 @@
+        The overwrites that should be applied to the dataset.
+
+    Returns
+    -------
+    adjustments : dict[idx -> Float64Overwrite]
+        The adjustments dictionary to feed to the adjusted array.
+    """
+    ad_series = deltas.loc[:, AD_FIELD_NAME]
+    return {
+        dates.get_loc(kd): concat(tuple(
+            overwrite_from_dates(
+                ad_series.loc[kd],
+                dates,
+                dense_dates,
+                n,
+                v,
+            ) for n in range(len(assets)))
+        ) for kd, v in deltas[column_name].iteritems()
+    }
+
+
+def adjustments_from_deltas_with_sids(dates,
+                                      dense_dates,
+                                      column_idx,
+                                      column_name,
+                                      assets,
+                                      deltas):
+    """Collect all the adjustments that occur in a dataset that does not
+    have a sid column.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that were in the dense data.
+    column_idx : int
+        The index of the column in the dataset.
+    column_name : str
+        The name of the column to compute deltas for.
+    deltas : pd.DataFrame
+        The overwrites that should be applied to the dataset.
+
+    Returns
+    -------
+    adjustments : dict[idx -> Float64Overwrite]
+        The adjustments dictionary to feed to the adjusted array.
+    """
+    ad_series = deltas[AD_FIELD_NAME]
+    adjustments = defaultdict(list)
+    for sid_idx, (sid, per_sid) in enumerate(deltas[column_name].iteritems()):
+        for kd, v in per_sid.iteritems():
+            adjustments[dates.get_loc(kd)].extend(
+                overwrite_from_dates(
+                    ad_series.loc[kd, sid],
+                    dates,
+                    dense_dates,
+                    sid_idx,
+                    v,
+                ),
+            )
+    return dict(adjustments)  # no subclasses of dict
+
+
+class BlazeLoader(dict):
+    def __init__(self, colmap=None):
+        self.update(colmap or {})
+
+    @classmethod
+    @memoize(cache=WeakKeyDictionary())
+    def global_instance(cls):
+        return cls()
+
+    def __hash__(self):
+        return id(self)
+
+    def __call__(self, column):
+        if column.dataset in self:
+            return self
+        raise KeyError(column)
+
+    def load_adjusted_array(self, columns, dates, assets, mask):
+        return map(
+            op.getitem(
+                dict(concat(
+                    self._load_dataset(cs, dates, assets, mask)
+                    for _, cs in iteritems(groupby(getdataset, columns))
+                )),
+            ),
+            columns,
+        )
+
+    def _load_dataset(self, columns, dates, assets, mask):
+        try:
+            dataset, = set(map(getdataset, columns))
```

Comment:
It's very easy to miss the comma here.  Can we put this in parens, or just do `if len(dataset) > 1`?